actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -128,18 +128,21 @@ def stylesheet_link_tag(*sources)
       #   # => <link rel="alternate" type="application/rss+xml" title="RSS" href="http://www.currenthost.com/news/feed" />
       #   auto_discovery_link_tag(:rss, "http://www.example.com/feed.rss", {title: "Example RSS"})
       #   # => <link rel="alternate" type="application/rss+xml" title="Example RSS" href="http://www.example.com/feed" />
+      #   auto_discovery_link_tag(:atom, {controller: "news", action: "feed"}, {data: {'turbolinks-track' => true}})
+      #   # => <link rel="alternate" type="application/atom+xml" title="ATOM" href="http://www.currenthost.com/news/feed" data-turbolinks-track="true" />
       def auto_discovery_link_tag(type = :rss, url_options = {}, tag_options = {})
         if !(type == :rss || type == :atom) && tag_options[:type].blank?
           raise ArgumentError.new("You should pass :type tag_option key explicitly, because you have passed #{type} type other than :rss or :atom.")
         end
 
-        tag(
-          "link",
+        tag_options.merge!({
           "rel"   => tag_options[:rel] || "alternate",
           "type"  => tag_options[:type] || Mime::Type.lookup_by_extension(type.to_s).to_s,
           "title" => tag_options[:title] || type.to_s.upcase,
```

Comment:
now that we merge the whole passed hash, are these conditionals still required? Wouldn't passed options simply overwrite existing defaults?

Maybe we need to look into `HashWithIndifferentAccess` to make symbol and string work the same way.